ecosystem/indexer-grpc/indexer-grpc-file-store/src/processor.rs

```diff
@@ -81,145 +79,194 @@ impl Processor {
+                    Ok(mut res) => {
+                        // Check for gaps
+                        res.sort_by(|a, b| a.0.cmp(&b.0));
+                        let mut prev_start = None;
+                        let mut prev_end = None;
+
+                        let first_version = res.first().unwrap().0;
+                        let last_version = res.last().unwrap().1;
+                        let first_version_encoded = res.first().unwrap().2.clone();
+                        let last_version_encoded = res.last().unwrap().2.clone();
+                        let versions: Vec<u64> = res.iter().map(|x| x.0).collect();
+                        for result in res {
+                            let start = result.0;
+                            let end = result.1;
+                            if prev_start.is_none() {
+                                prev_start = Some(start);
+                                prev_end = Some(end);
+                            } else {
+                                if prev_end.unwrap() + 1 != start {
+                                    tracing::error!(
+                                        processed_versions = ?versions,
+                                        "[Filestore] Gaps in processing data"
+                                    );
+                                    panic!("[Filestore] Gaps in processing data");
+                                }
+                                prev_start = Some(start);
+                                prev_end = Some(end);
+                            }
+                        }
+
+                        (
+                            first_version,
+                            last_version,
+                            first_version_encoded,
+                            last_version_encoded,
+                        )
+                    },
+                    Err(err) => panic!("Error processing transaction batches: {:?}", err),
+                };
 
-            // Drain the transactions buffer and upload to file store in size of multiple of BLOB_STORAGE_SIZE.
-            let process_size = transactions_buffer.len() / BLOB_STORAGE_SIZE * BLOB_STORAGE_SIZE;
-            let current_batch: Vec<EncodedTransactionWithVersion> =
-                transactions_buffer.drain(..process_size).collect();
-            let last_transaction = current_batch.as_slice().last().unwrap().clone();
-            let first_transaction = current_batch.as_slice().first().unwrap().clone();
-            self.file_store_operator
-                .upload_transactions(cache_chain_id, current_batch)
+            // update next batch start version
+            batch_start_version = last_version + 1;
+            assert!(
+                batch_start_version % BLOB_STORAGE_SIZE as u64 == 0,
+                "[Filestore] Batch must be multiple of 1000"
+            );
+            let size = last_version - first_version + 1;
+            PROCESSED_VERSIONS_COUNT.inc_by(size);
```

Comment:
can you use NUM_TRANSACTIONS metrics instead? that is the new metric. `PROCESSED_VERSIONS_COUNT` will be deprecated

Reply:
It's in log_grpc_step already.